### ISSUE TITLE
Win32 improvements for MyThes

### ISSUE THREAD

**Issue by phajdan** [hunspell/hunspell#329]
\- simplified building dll
\- extended API for work with UTF-8 strings on Windows.

Original comment by: *anonymous

Original Ticket: [hunspell/patches/48](https://sourceforge.net/p/hunspell/patches/48)

**Comment by dimztimz** [comment]
In windows dealing with filenames in UTF-8 is useless. This patch wont fix things in general case.

The windows NT kernel and the NTFS deal exclusively with UTF-16 strings.

Let's say we pass the filename as command line argument. Then inside the C, we can read the arguments in two ways:

- as wide char strings (UTF-16) using `GetCommandLine` function. This is the proper way.
- as single byte encoding by just reading `char * argv[]`. What internally happens is that `argv` is populated by converting the UTF-16 command line to some ANSI encoding that is set in the regional settings of control panel as "locale for non Unicode applications".
So, `char * argv` is done by UTF-16 to ANSI.

And what happens when we use fopen(char *).
I guess MSVCRT expects that the char * string is encoded in ANSI and does ANSI to UTF-16 before actually opening the file.

If we want no conversions and full unicode support, on windows one must use:
`wchar_t * argv` for getting a filename, and `_wfopen` for opening.

Basically there are two ways for handing filenames with international characters in Windows.

1. Just use the standard `char * argv` and` fopen` and hope that the right codepage is set in Control panel. OR
2. Use platform specific `wchar_t * argv` and `_wfopen`.
